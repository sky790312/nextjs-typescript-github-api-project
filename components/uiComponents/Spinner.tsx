import { FunctionComponent } from 'react'
import styled from 'styled-components'

export const Spinner: FunctionComponent = () => <StyledSpinner />

const StyledSpinner = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  width: 2rem;
  height: 2rem;
  margin: -1rem 0 0 -1rem;
  border-radius: 50%;
  border: 0.3rem solid;
  border-color: #d8d8d8;
  border-top-color: #fff;
  animation: 1.5s spin infinite linear;

  @keyframes spin {
    to {
      transform: rotate(360deg);
    }
  }
`
