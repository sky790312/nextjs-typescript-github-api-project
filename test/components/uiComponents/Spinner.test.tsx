import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import { Spinner } from '@/components/uiComponents/Spinner'

describe('Spinner ui component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Spinner />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
