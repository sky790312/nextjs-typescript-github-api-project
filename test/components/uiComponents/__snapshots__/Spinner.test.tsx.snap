// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner ui component matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="Spinner__StyledSpinner-gyn5vx-0 lkOToq"
  />
</DocumentFragment>
`;
